cmd/kubectl-gadget/top/tcp.go

```diff
@@ -15,63 +15,53 @@
 package top
 
 import (
-	"encoding/json"
 	"fmt"
-	"os"
 	"strconv"
 	"strings"
-	"sync"
 	"syscall"
-	"time"
 
 	"github.com/spf13/cobra"
-	"golang.org/x/term"
 
 	commonutils "github.com/inspektor-gadget/inspektor-gadget/cmd/common/utils"
 	"github.com/inspektor-gadget/inspektor-gadget/cmd/kubectl-gadget/utils"
-	gadgetv1alpha1 "github.com/inspektor-gadget/inspektor-gadget/pkg/apis/gadget/v1alpha1"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/top"
 	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/top/tcp/types"
 )
 
-type TCPFlags struct {
-	CommonTopFlags
-
-	ParsedSortBy types.SortBy
-	FilteredPid  uint
-	Family       uint
-}
-
 type TCPParser struct {
 	commonutils.BaseParser[types.Stats]
-	sync.Mutex
 
-	flags     *TCPFlags
-	nodeStats map[string][]types.Stats
+	flags *CommonTopFlags
 }
 
 func newTCPCmd() *cobra.Command {
-	var flags TCPFlags
-
-	commonFlags := &utils.CommonFlags{
-		OutputConfig: commonutils.OutputConfig{
-			// The columns that will be used in case the user does not specify
-			// which specific columns they want to print.
-			CustomColumns: []string{
-				"node",
-				"namespace",
-				"pod",
-				"container",
-				"pid",
-				"comm",
-				"ip",
-				"saddr",
-				"daddr",
-				"sent",
-				"received",
+	commonTopFlags := &CommonTopFlags{
+		CommonFlags: utils.CommonFlags{
+			OutputConfig: commonutils.OutputConfig{
+				// The columns that will be used in case the user does not specify
+				// which specific columns they want to print.
+				CustomColumns: []string{
+					"node",
+					"namespace",
+					"pod",
+					"container",
+					"pid",
+					"comm",
+					"ip",
+					"saddr",
+					"daddr",
+					"sent",
+					"received",
+				},
 			},
 		},
 	}
 
+	var (
+		FilteredPid uint
+		Family      uint
```

Comment:
nit: Usually those variable start with lower case.

```suggestion
		filteredPid uint
		family      uint
```